### PR TITLE
MJCF to SDF: Added geometry defaults

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>
# 🎉 New feature

## Summary

You can define some defaul values in mujoco:

```xml
   <default>
     <geom type="capsule" mass="1"/>
   </default>
```

This PR allows to read these values and set them

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by azeey — CHANGES_REQUESTED** [review]
I've left a comment on the logic of processing default classes. My preference would be to put all the default class processing into a single
module instead of duplicating them in each conversion file so that we only have one place to fix if we find bugs down the road.

File: mjcf_to_sdformat/tests/resources/test_mujoco.xml [hunk 1]
```diff
@@ -1,4 +1,8 @@
 <mujoco>
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply mjcf_to_sdformat/tests/resources/test_mujoco.xml:1]: https://github.com/gazebosim/gz-mujoco/pull/43/commits/7e508b314f542f9e04de279dc4a740e2fadd7031

File: mjcf_to_sdformat/mjcf_to_sdformat/converters/link.py [hunk 2]
```diff
@@ -27,12 +27,49 @@
...
+                    except KeyError:
+                        geom.set_attributes(**{k: v})
+    if geom.dclass is not None:
+        if geom.dclass.geom is not None:
+            for k, v in geom.dclass.geom.get_attributes().items():
+                try:
+                    geom.get_attributes()[k]
+                    geom.set_attributes(**{k: v})
+                except KeyError:
+                    geom.set_attributes(**{k: v})
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply mjcf_to_sdformat/mjcf_to_sdformat/converters/link.py]: https://github.com/gazebosim/gz-mujoco/pull/43/commits/7e508b314f542f9e04de279dc4a740e2fadd7031

File: mjcf_to_sdformat/mjcf_to_sdformat/converters/world.py [hunk 3]
```diff
@@ -48,11 +48,21 @@ def mjcf_worldbody_to_sdf(mjcf_root, world):
 
     body = mjcf_root.worldbody.body
 
-    def iterate_bodies(input_body, model, body_parent_name=None):
+    def iterate_bodies(input_body,
+                       model,
+                       body_parent_name=None,
+                       default_classes=[]):
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply mjcf_to_sdformat/mjcf_to_sdformat/converters/world.py]: https://github.com/gazebosim/gz-mujoco/pull/43/commits/7e508b314f542f9e04de279dc4a740e2fadd7031

File: mjcf_to_sdformat/tests/resources/test_defaults.xml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+<mujoco>
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply mjcf_to_sdformat/tests/resources/test_defaults.xml:1]: https://github.com/gazebosim/gz-mujoco/pull/43/commits/7e508b314f542f9e04de279dc4a740e2fadd7031

File: mjcf_to_sdformat/mjcf_to_sdformat/converters/link.py [hunk 5]
```diff
@@ -27,12 +27,49 @@
...
+    to the body and parent bodies
+    :return: The modfied MJCF geometry.
+    :rtype: mjcf.Element
+    """
+    if geom.root.default.geom is not None:
+        for k, v in geom.root.default.geom.get_attributes().items():
+            try:
+                geom.get_attributes()[k]
+            except KeyError:
+                geom.set_attributes(**{k: v})
```
[the remark this answers is not in the document]
Comment (ahcorde) [reply mjcf_to_sdformat/mjcf_to_sdformat/converters/link.py]: https://github.com/gazebosim/gz-mujoco/pull/43/commits/7e508b314f542f9e04de279dc4a740e2fadd7031

[the remark this answers is not in the document]
Comment (azeey) [reply mjcf_to_sdformat/mjcf_to_sdformat/converters/link.py]: Also addressed by #61

File: mjcf_to_sdformat/mjcf_to_sdformat/converters/link.py [hunk 6]
```diff
@@ -27,12 +27,49 @@
...
+
+    if default_classes is not None:
+        for default_class in default_classes:
+            if default_class.geom is not None:
+                for k, v in default_class.geom.get_attributes().items():
+                    try:
+                        geom.get_attributes()[k]
+                    except KeyError:
+                        geom.set_attributes(**{k: v})
+    if geom.dclass is not None:
```
[the remark this answers is not in the document]
Comment (azeey) [reply mjcf_to_sdformat/mjcf_to_sdformat/converters/link.py]: Also addressed by #61

**Review by azeey — APPROVED** [review]

**Pull request merged**